### ISSUE TITLE
Transaction report missing added or retracted flag

### ISSUE THREAD

**Issue by TimoKramer** [replikativ/datahike#243]
I expect the transaction report to show which datoms are added or retracted. It does show the datoms multiple times but not what happened to them.
```
(d/transact db [[:db/add 1, :name "Ivan"]
                                         [:db/add 1, :likes "fries"]
                                         [:db/add 1, :likes "pizza"]
                                         [:db/add 1, :friends "Olaf"]
                                         [:db/add 1, :friends "Gerd"]
                                         [:db/add 1, :friends "Toni"]
                                         [:db/add 1, :friends "Sergey"]])
```
```
{:db-before #datahike/DB {:max-tx 536870913 :max-eid 1},
 :db-after #datahike/DB {:max-tx 536870914 :max-eid 1},
 :tx-data
 [#datahike/Datom [1 :likes "pizza" 536870914]
  #datahike/Datom [1 :likes "fries" 536870914]
  #datahike/Datom [1 :likes "fries" 536870914]
  #datahike/Datom [1 :likes "pizza" 536870914]
  #datahike/Datom [1 :friends "Sergey" 536870914]
  #datahike/Datom [1 :friends "Olaf" 536870914]
  #datahike/Datom [1 :friends "Olaf" 536870914]
  #datahike/Datom [1 :friends "Gerd" 536870914]
  #datahike/Datom [1 :friends "Gerd" 536870914]
  #datahike/Datom [1 :friends "Toni" 536870914]
  #datahike/Datom [1 :friends "Toni" 536870914]
  #datahike/Datom [1 :friends "Sergey" 536870914]],
 :tempids #:db{:current-tx 536870914},
 :tx-meta nil}
```
Obviously in this case it shouldn't be retracted at all. This is the issue #242.

**Comment by whilo** [comment]
We cannot retain the report as we did before, because upserts now happen lazily and will only be applied once they hit the data node in the hitchhiker-tree (which also happens when you read). We could add the upserts to the report though @grischoun . @TimoKramer What do you need the report for?

**Comment by TimoKramer** [comment]
When I was writing tests for the api-ns overhaul I tried all the examples from the docs and could not reproduce the ones for the datoms api. So I wrote my own tests with schema and inputs. For the datoms api it was really hard for me to find an input that reproduces the output from the previous examples. (https://github.com/replikativ/datahike/blob/745133fa2240771dee2bfffc947b4db48c74549e/test/datahike/test/api.cljc#L402)

Complicating was the output of multiple datoms in the transaction report without a transact/retract-flag that made it even harder to understand what was going on. It was @jsmassa who mentioned that there is supposed to be a transact/retract flag and @kordano also was not aware that this is not possible anymore to show.
